backup/restore

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env bash
+
+## Restore from a backup
+set -eE
+
+volumes=("hint_db_data" "hint_redis_data" "hint_prerun" "hint_results" "hint_uploads" "hint_config")
+
+array_contains() {
+    local seeking=$1
+    shift
+    local in=1
+    for element; do
+        if [[ $element == "$seeking" ]]; then
+            in=0
+            break
+        fi
+    done
+    return $in
+}
+
+assert_file_exists() {
+    if !([ ! -z "$1" ] && [ -f "$1" ]); then
```

Comment:
I just got bitten by `-f $path` returning false for a directory and `-d $path` was needed - does this work as expected?